pepper-apis/src/main/java/org/broadinstitute/ddp/util/GsonRecursiveReader.java

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.ddp.util;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+/**
+ * Provides reading of node values from JSON doc.<br>
+ * Main method: {@link #readValues(JsonElement, List)}<br>
+ * Example:
+ * <pre>
+ *   var result = GsonRecursiveReader.getValues(rootElem, List.of("log_id", "date", "type"));
+ * </pre>
+ * The result will contain map with elements found by names (specified by list of strings).
+ * Each name searched recursively: if it is not found at top level then try to find it deeper etc..
+ * As soon as a name is found - it stopped to search this name in the document.
+ */
+public class GsonRecursiveReader {
+
+    /**
+     * Main method providing reading of {@link JsonElement}-s by it's names.
+     * The elements searched recursively (from top to bottom) and as soon as a name found
+     * it stops searching of this name (i.e. more deeper elements with the same name not override
+     * already detected element).
+     *
+     * @param rootNode root element of JSON document where to find elements
+     * @param names    list of elements' names which to search for
+     * @return {@code Map<String, JsonElement>} map with detected node values
+     */
+    public static Map<String, JsonElement> readValues(final JsonElement rootNode, final List<String> names) {
+        final Map<String, JsonElement> results = new HashMap<>();
+        if (names != null && names.size() > 0) {
+            final List<String> yetNotReadNames = new ArrayList<>(names);
```

Comment:
Are we using a `List` because we want to parse the names in the given order? Curious why not use a `Set`?

Reply:
It just a temp List: intitally contains all names and as soon as certain names found in JSON it removed from the list (so we not going to search it further in the JSON). I changed ArrayList to LinkedList for better performance of removing.

Reply:
I asked because if order doesn't matter then I would've suggested using `Set`. But this is a small enough thing compared to other things going on in this PR, so let's close this topic for now.